src/test/java/CarComfortServiceTest.java

```diff
@@ -0,0 +1,243 @@
+        assertNotNull(carComfortService);
+    }
+
+
+    @Test
+    public void whenInsertCorrectCarComfort_thenVerifyCarComfort() {
+        CarComfort carComfort = buildCarComfort(UUID.randomUUID());
+
+        when(carComfortRepositoryMock.insert(carComfort)).thenReturn(carComfort);
+
+        CarComfort response = carComfortService.insert(carComfort);
+        assertNotNull(response);
+
+        assertEquals(carComfort.getId(), response.getId());
+        assertEquals(carComfort.getName(), response.getName());
+        assertEquals(carComfort.getDescription(), response.getDescription());
+
+        verify(carComfortRepositoryMock).insert(carComfort);
+        verifyNoMoreInteractions(carComfortRepositoryMock);
+    }
+
+
+    @Test
+    public void whenInsertNullCarComfort_thenVerifyException() {
+        String exceptionMessage = "Car comfort must be NOT null";
+
+        CarRentalException exception = assertThrows(
+                CarRentalException.class, () -> carComfortService.insert(null));
+        assertNotNull(exception);
+        assertEquals(exceptionMessage, exception.getMessage());
+
+        verifyNoInteractions(carComfortRepositoryMock);
+    }
+
+
+    @Test
+    public void whenInsertInvalidCarComfort_thenVerifyException() {
+        CarComfort carComfort = new CarComfort();
+
+        String exceptionMessage = "Car comfort has invalid data";
+        CarRentalException exception = assertThrows(
+                CarRentalException.class, () -> carComfortService.insert(carComfort));
+        assertNotNull(exception);
+        assertEquals(exceptionMessage, exception.getMessage());
+
+        verifyNoInteractions(carComfortRepositoryMock);
+    }
+
+
+    @Test
+    public void whenGetAll_thenVerifyListOfCarComforts() {
+        List<CarComfort> carComforts = List.of(
+                buildCarComfort(UUID.randomUUID()),
+                buildCarComfort(UUID.randomUUID()),
+                buildCarComfort(UUID.randomUUID()));
+
+        when(carComfortRepositoryMock.getAll()).thenReturn(carComforts);
+
+        List<CarComfort> response = carComfortService.getAll();
+        assertNotNull(response);
+        assertEquals(carComforts.size(), response.size());
+        assertArrayEquals(carComforts.toArray(), response.toArray());
+
+        verify(carComfortRepositoryMock).getAll();
+        verifyNoMoreInteractions(carComfortRepositoryMock);
+    }
+
+
+    @Test
+    public void whenGetByCorrectId_thenVerifyCarComfort() {
+        UUID id = UUID.randomUUID();
+        CarComfort carComfort = buildCarComfort(UUID.randomUUID());
+
+        when(carComfortRepositoryMock.getById(id)).thenReturn(carComfort);
+
+        CarComfort response = carComfortService.getById(id);
+        assertNotNull(response);
+        assertEquals(carComfort.getId(), response.getId());
+        assertEquals(carComfort.getName(), response.getName());
+        assertEquals(carComfort.getDescription(), response.getDescription());
```

Comment:
You can use assertEquals(carComfort, response) as above

Reply:
Done